File: Data Structures & Algorithms/sum-of-all-subset-xor-totals/submission-1.java
class Solution {
    int res = 0;
    public int subsetXORSum(int[] nums) {
        backtrack(0, 0, nums);
        return res;
    }

    public void backtrack(int index, int currXorr, int[] nums) {
        res += currXorr;

        for (int i = index; i < nums.length; i++) {
            backtrack(i + 1, currXorr ^ nums[i], nums);
        }
    }
}

File: Data Structures & Algorithms/subsets/submission-1.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        res = []
        
        def dfs(index, curr):
            res.append(curr.copy())

            for i in range(index, len(nums)):
                curr.append(nums[i])
                dfs(i + 1, curr)
                curr.pop()

        dfs(0, [])
        return res
        